code/game/objects/items/weapons/grenades/atmosgrenade.dm

```diff
@@ -0,0 +1,98 @@
+
+
+
+/obj/item/weapon/grenade/gas
+	name = "Plasma Fire Grenade"
+	desc = "A compressed plasma grenade, used to start horrific plasma fires."
+	origin_tech = "materials=3;magnets=4;syndicate=4"
+	icon = 'icons/obj/grenade.dmi'
+	icon_state = "syndicate"
+	item_state = "flashbang"
+	var/gas_type = "fire"
+	var/max_spread_turfs = 20
+	var/gas_debug = 0
+
+/obj/item/weapon/grenade/gas/overpressure
+	name = "Overpressure Grenade"
+	desc = "A compressed gas grenade, used to create crushing pressure in an area."
+	gas_type = "overpressure"
+
+/obj/item/weapon/grenade/gas/suffocation
+	name = "Suffocation Grenade"
+	desc = "An oxygen depletion grenade, used to ensure anyone entering the target area quickly suffocates."
+	gas_type = "suffocation"
+
+/obj/item/weapon/grenade/gas/vacuum
+	name = "Vacuum Grenade"
+	desc = "A grenade that completely removes all air and heat from its detonation area."
+	gas_type = "vacuum"
+
+/obj/item/weapon/grenade/gas/knockout
+	name = "Knockout Grenade"
+	desc = "A grenade that completely removes all air and heat from its detonation area."
+	gas_type = "knockout"
+
+/obj/item/weapon/grenade/gas/prime()
+	var/turf/simulated/origin = get_turf(src)
+	var/list/spread_turfs = list()
+	var/list/scan_turfs = list()
+
+	if(!origin)
+		// Error, unsimulated turf.
+		return
```

Comment:
This won't stop it from working on an unsimulated turf - you'll want an `istype` for that